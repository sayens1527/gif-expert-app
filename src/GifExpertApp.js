import React, { useState } from 'react'
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {

    let [categories, setCategories] = useState(defaultCategories);

    /*const hadleAdd = () => {
        setCategories([...categories,'Nueva categoria']);
    };*/

    return (
        <div>
            <h2>Gif expert App</h2>

            <AddCategory setCategories = { setCategories }></AddCategory>

            <hr/>

            <ol>
                { categories.map( cat => (
                    <GifGrid
                        key={cat}
                        category={cat}
                    />
                ))}
            </ol>
        </div>
    )
}

export default GifExpertApp
        