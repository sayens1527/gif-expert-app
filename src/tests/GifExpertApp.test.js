import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import GifExpertApp from '../GifExpertApp';
import { expect } from '@jest/globals';


describe('Test in GifExpertApp component', () => {
    
    test('should match with snapshot', () => {

        const wraper = shallow(<GifExpertApp></GifExpertApp>);

        expect(wraper).toMatchSnapshot();
        
    });

    test('should show a category list', () => {

        const categories= ['1','2','3'];
        
        const wraper = shallow(<GifExpertApp defaultCategories = {categories}></GifExpertApp>);

        const gridGifs = wraper.find('GifGrid').length;
        
        expect(gridGifs).toBe(categories.length);

    });
    

    test('should has  Add category component', () => {
        
        const wraper = shallow(<GifExpertApp></GifExpertApp>);

        const addCategoryElement = wraper.find('AddCategory');

        expect(addCategoryElement.exists()).toBe(true);

    });
    

});
