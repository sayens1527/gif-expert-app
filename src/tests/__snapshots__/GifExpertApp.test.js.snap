// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in GifExpertApp component should match with snapshot 1`] = `
<div>
  <h2>
    Gif expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
